tests/erlang_tests/test_base64.erl

```diff
@@ -69,13 +69,26 @@ start() ->
     expect_error(fun() -> base64:decode(<<"A ID">>) end, badarg),
 
     % it turns out we actually support iolists, which is kind of nice
-    <<"AQIDBAUG">> = base64:encode([<<1, 2>>, <<3, 4, 5>>, <<6>>]),
-    <<"AQIDBAUG">> = base64:encode([<<1, 2>>, <<3, 4, 5>>, 6]),
-    <<1, 2, 3, 4, 5, 6>> = base64:decode(["AQ", "ID", "BAUG"]),
-    <<1, 2, 3, 4, 5, 6>> = base64:decode(["AQ", "ID", <<"BAUG">>]),
-
+    case erlang:system_info(machine) of
+        "BEAM" ->
+            expect_error(fun() -> base64:encode([<<1, 2>>, <<3, 4, 5>>, <<6>>]) end, badarith),
+            expect_error(fun() -> base64:encode([<<1, 2>>, <<3, 4, 5>>, <<6>>]) end, badarith),
+            expect_error(fun() -> base64:decode(["AQ", "ID", "BAUG"]) end, badarg),
+            expect_error(fun() -> base64:decode(["AQ", "ID", <<"BAUG">>]) end, badarg);
+        _ ->
+            <<"AQIDBAUG">> = base64:encode([<<1, 2>>, <<3, 4, 5>>, <<6>>]),
+            <<"AQIDBAUG">> = base64:encode([<<1, 2>>, <<3, 4, 5>>, 6]),
+            <<1, 2, 3, 4, 5, 6>> = base64:decode(["AQ", "ID", "BAUG"]),
+            <<1, 2, 3, 4, 5, 6>> = base64:decode(["AQ", "ID", <<"BAUG">>])
+    end,
```

Comment:
This is interesting.  I took a closer look at the docs and OTP does not support base-64 encoding of io-lists.  But I think for memory-constrained systems I think that is a pretty important feature.  So I agree with the way you have constructed this test.